Tests/NIOHTTP2Tests/HTTP2StreamMultiplexerTests.swift

```diff
@@ -91,13 +92,16 @@ typealias FramePayloadWriteRecorder = WriteRecorder<HTTP2Frame.FramePayload>
 
 
 /// A handler that keeps track of all reads made on a channel.
-final class InboundRecorder<Frame>: ChannelInboundHandler {
+final class InboundRecorder<Frame>: ChannelInboundHandler, @unchecked Sendable {
     typealias InboundIn = Frame
 
+    let framesLock = NIOLock()
     var receivedFrames: [Frame] = []
 
     func channelRead(context: ChannelHandlerContext, data: NIOAny) {
-        self.receivedFrames.append(self.unwrapInboundIn(data))
+        self.framesLock.withLock {
+            self.receivedFrames.append(self.unwrapInboundIn(data))
+        }
```

Comment:
This is not _quite_ right. 

`receivedFrames` is still accessible. Our normal convention is to rename the protected state to be prefixed with `_` and then have a whatever-access-level-is-approproate computed property which acquires and releases the lock when accessing the protected state. Within the object holding the state and the lock you can manually acquire it if, for example, you need finer grained lock control (i.e. it protects more than one piece of state and you need to modify both).